src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```
@@ -1268,6 +1287,83 @@ private static JsonNode parseCommandInGS(SFStatement statement, String command)
     return result;
   }
 
+  /**
+   * This is API function to parse the File Transfer Metadatas from a supplied PUT call response.
+   *
+   * <p>NOTE: It only supports PUT on S3/AZURE/GCS
+   *
+   * @param jsonNode JSON doc returned by GS from PUT call
+   * @return The file transfer metadatas for to-be-transferred files.
+   * @throws SnowflakeSQLException if any error occurs
+   */
+  public static List<SnowflakeFileTransferMetadata> getFileTransferMetadatas(JsonNode jsonNode)
+      throws SnowflakeSQLException {
+    CommandType commandType = CommandType.UPLOAD;
+    if (!jsonNode.path("data").path("command").isMissingNode()) {
+      commandType = CommandType.valueOf(jsonNode.path("data").path("command").asText());
+    }
+
+    if (commandType != CommandType.UPLOAD) {
+      throw new SnowflakeSQLException(
+          new Exception(),
+          SqlState.INTERNAL_ERROR,
+          ErrorCode.INTERNAL_ERROR.getMessageCode(),
+          "This API only supports PUT command");
+    }
+
+    JsonNode locationsNode = jsonNode.path("data").path("src_locations");
+
+    assert locationsNode.isArray();
```

Can we return an exception here if this is not an array?